### PR TITLE
feat(cursor): improve theme styling for crosshair

### LINKED ISSUE #925: Crosshair line for Line / Area charts is too light
**Describe the bug**

The crosshair line for Line/Area charts is too light. There is an open discussion in EUI (https://github.com/elastic/eui/issues/4308).

**To Reproduce**

1. Go to https://elastic.github.io/elastic-charts/?path=/story/line-chart--basic
2. Move the mouse over the line chart
4. The crosshair line is too light 

**Expected behaviour**

The expected behaviour is that the line charts get the `crosshair.line` styles and right now it is getting the `crosshair.band` styles which are too light in color:

https://github.com/elastic/eui/blob/c0b8a25095e9fd60a5a9911d0cb1ba1249400ebf/src/themes/charts/themes.ts#L172-L181

**Screenshots**

Screenshot 2020-11-25 at 12 49 11
 

**Kibana Cross Issues**

This issue was noticed while working on:
https://github.com/elastic/kibana/pull/81565

### PR DESCRIPTION
**Pull request by markov00**
## Summary

This PR improved the styling for the crosshair cursor:
- the `crosshair` theme has a more specific `crossLine` styling object that is used style the horizontal line when using a `TooltipType.Crosshairs`
```ts
interface CrosshairStyle {
  band: FillStyle & Visible;
  line: StrokeStyle & Visible & Partial<StrokeDashArray>;
  crossLine: StrokeStyle & Visible & Partial<StrokeDashArray>;
}
```
- the "vertical" now correctly pick the style from the theme depending on its shape (using the `band` style when the chart has at least one bar series, using the `line` style when the chart is composed of line or/and areas) and using the `crossLine` to style the horizontal crossing line

Screenshot 2021-01-15 at 10 29 08

- The cursor implementation was also refactored using SVG instead of DIV elements allowing us to correctly align the cursors to the underlying scale, fixing the current misalignment visible in various cases

Before and after
Screenshot 2021-01-15 at 10 21 52Screenshot 2021-01-15 at 10 21 45

fix #925

### CODE REVIEW

**Comment by markov00** [comment]
Hey @miukimiu could you please take a look at the default color I've used here for the line cursor?
I think we should update the EUI theme to use the right color for the `crosshair.line` to have a smooth transition when updating both libraries in kibana
You can test the default color assigned here in the local storybook at `http://localhost:9001/?path=/story/stylings--dark-theme` and playing with dark mode or light mode and with or without bars

File: src/chart_types/xy_chart/crosshair/crosshair_utils.ts [hunk 1]
```diff
@@ -96,31 +97,22 @@ export function getCursorBandPosition(
...
+
   if (x > chartWidth || y > chartHeight || x < 0 || y < 0 || !invertedValue.withinBandwidth) {
-    return {
-      top: -1,
-      left: -1,
-      width: 0,
-      height: 0,
-      visible: false,
-    };
+    return undefined;
```
Comment (nickofthyme) [inline src/chart_types/xy_chart/crosshair/crosshair_utils.ts:110]: 👍

File: src/chart_types/xy_chart/renderer/dom/_crosshair.scss [hunk 2]
```diff
@@ -1,6 +1,6 @@
 .echCrosshair,
-.echCrosshair__band,
-.echCrosshair__line {
+.echCrosshair__cursor,
+.echCrosshair__crossLine {
```
Comment (nickofthyme) [inline src/chart_types/xy_chart/renderer/dom/_crosshair.scss:3]: ❤️

**Review by nickofthyme — APPROVED** [review]
LGTM, tested locally. This looks so much nicer and cleaner using the svg instead 👍

**Review by rshen91 — APPROVED** [review]
LGTM!

File: src/utils/themes/light_theme.ts [hunk 3]
```diff
@@ -176,6 +176,11 @@ export const LIGHT_THEME: Theme = {
       visible: true,
     },
     line: {
+      stroke: '#98A2B3',
+      strokeWidth: 1,
+      visible: true,
+    },
+    crossLine: {
       stroke: '#777',
```
Comment (miukimiu) [inline src/utils/themes/light_theme.ts:184]: The `line.stroke` and `crossLine.stroke` could use the same color. I think the `#98A2B3` has more contrast. 

```suggestion
      stroke: '#98A2B3',
```

File: src/utils/themes/dark_theme.ts [hunk 4]
```diff
@@ -176,6 +176,11 @@ export const DARK_THEME: Theme = {
       visible: true,
     },
     line: {
+      stroke: '#535966',
```
Comment (miukimiu) [inline src/utils/themes/dark_theme.ts:179]: The `line.stroke` and `crossLine.stroke` could use the same color. And `#999` has more contract on darker backgrounds:

```suggestion
      stroke: '#999',
```

**Review by miukimiu — COMMENTED** [review]
LGTM! 🎉 

I tested locally and just added a few suggestions. I can update the EUI theme once the PR is merged.

File: src/utils/themes/light_theme.ts (hunk 3 above)
Comment (nickofthyme) [reply src/utils/themes/light_theme.ts:184]: Just to note: these styles are only default to elastic charts. If you want this change reflected in kibana you'd need to update the eui charts theme.

https://github.com/elastic/eui/blob/d16d296ae4d06a6337a4a2534f4d3bfacff932ea/src/themes/charts/themes.ts#L172-L181

File: src/utils/themes/dark_theme.ts (hunk 4 above)
Comment (markov00) [reply src/utils/themes/dark_theme.ts]: done in #5e8f506

File: src/utils/themes/light_theme.ts (hunk 3 above)
Comment (markov00) [reply src/utils/themes/light_theme.ts]: done in #5e8f506

**Pull request merged**